### DISCUSSION TITLE
What to use, LIN_ADVANCE or S_CURVE_ACCELERATION?

### DISCUSSION THREAD

**Discussion by matthiasp101** [Jyers/Marlin#1628] [Q&A]
I have Ender 3 v2 with default Creality Motherboard v4.2.2.

What should I use, LIN_ADVANCE or S_CURVE_ACCELERATION?

**Comment by Cal-LF** [comment]
Linear advance does not work unless the board is modified.

**Reply by Silenstryke** [reply]
The 4.2.2 mobo has multiple revisions, indicated by a letter marked on the card reader. Source: https://github.com/Jyers/Marlin/discussions/707#discussioncomment-701286

Its basically a lottery if your ender 3v2 with a 4.2.2 mobo has 2208s or 2209s because both revisions are still being sold. If card reader is marked with a A, C or E then you have to do a hardware mod to enable spreadcycle. The B revision has 2209s and LIN_ADVANCE seems to work without issue.

**Reply by Silenstryke** [reply]
Also discussed here on reddit: https://www.reddit.com/r/ender3/comments/mhahoc/the_mystery_of_the_linear_advance_lottery_of_the/

**Reply by Silenstryke** [reply]
@tititopher68-dev 
2209s can do LIN_ADVANCE in stealthchop mode, 2208 will not because of a flaw that has to do with EMF interference that was solved with the 2209. To get a 2208 to do LA requires spreadcycle, hence the hardware mod. Sources: https://github.com/Jyers/Marlin/discussions/707#discussioncomment-738599 https://github.com/Jyers/Marlin/discussions/707#discussioncomment-704832

The same issue is on the 4.2.7 boards with TMC2225 drivers, which are a re-packaged 2208. TMC2225 require pin 19 pulled high to manually toggle spreadcycle without using UART.

People have reported getting a 4.2.2 rev.B to do linear advance without mods.

**Comment by matthiasp101** [comment]
Ok, I have the motherboard with A on the sd card reader. I use S_CURVE_ACCELERATION now.

**Reply by Tapap84** [reply]
what is S_CURVE_ACCELERATION

**Reply by Tapap84** [reply]
im wanting to give this a try, would i do #define BEZIER_CURVE_SUPPORT  on line 2068? im using v2.0.1 source with 3x3 manual config
